src/main.ml

```diff
@@ -88,236 +96,232 @@ let reduce_string_table table width =
+                    if not !stop then (
+                      f ();
+                      Lwt_unix.sleep delay >>= step 0.025 )
+                    else Lwt.return_unit
+                  in
+                  Lwt.async (step 0.4);
+                  `Grab ((fun ~x:_ ~y:_ -> ()), fun ~x:_ ~y:_ -> stop := true)
+                in
+                Ui.mouse_area
+                  (fun ~x:_ ~y -> function
+                    | `Left ->
+                        if y < prefix then
+                          let position = ref st.position in
+                          repeat (fun () ->
+                              position := max 0 (!position - (st.visible / 2));
+                              set_scroll `Change
+                                { st with position = !position })
+                        else if y > prefix + ratio then
+                          let position = ref st.position in
+                          repeat (fun () ->
+                              position :=
+                                min st.bound (!position + (st.visible / 2));
+                              set_scroll `Change
+                                { st with position = !position })
+                        else
+                          `Grab
+                            ( (fun ~x:_ ~y:y' ->
+                                let dy = y' - y in
+                                let position =
+                                  float st.position
+                                  +. float dy /. float st.W.visible
+                                     *. float st.W.total
+                                in
+                                let position =
+                                  max 0 (min st.bound (int_of_float position))
+                                in
+                                set_scroll `Change { st with position }),
+                              fun ~x:_ ~y:_ -> () ) | _ -> `Unhandled)
+                  ( Ui.atom
+                  @@ Notty.I.vcat
+                       [
+                         bar gray prefix;
+                         bar lightgray ratio;
+                         bar gray suffix;
+                       ] )
+              else Ui.atom (bar gray st.W.visible))
+     in
+     let result =
+       Lwd_utils.pack Ui.pack_y
+         [
+           Lwd.pure (W.string text);
+           Lwd_utils.pack Ui.pack_x
+             [
+               Lwd.get buttons;
+               Lwd.pure
+                 (Ui.resize ~h:1 ~sw:1 ~bg:Notty.A.(bg (gray 1)) Ui.empty);
+             ];
+           Lwd_utils.pack Ui.pack_x
+             [
+               Lwd_utils.pack Ui.pack_y
+                 [ Lwd.bind (Lwd.get width) (reduce_string_table log_lines) ]
+               |> W.vscroll_area ~state:(Lwd.get scroll_state)
+                    ~change:set_scroll
+               |> Lwd.map (fun ui ->
+                      ui |> Ui.resize ~w:0 ~sw:1
+                      |> Ui.size_sensor (fun w _ -> update_if_changed width w));
```

Comment:
Another minor remark: `|>` are often put on different lines (to visually reflect the pipeline).
This seems to be the case elsewhere, but here only the second one is put on its own line (?!).

Reply:
Indeed, I think OCamlformat will essentially word-wrap them to 80 characters. I suspect the desired behaviour is to split them onto separate lines as soon as they spill over from a single line, but then there is the question of being consistent with user-defined sequencing like `>>=` and `>|=`.

Reply:
That's not great indeed.

Reply:
Wait ! There is an option for that. You could add `break-infix = fit-or-vertical` in `.ocamlformat`.